src/computer_session.cpp

```diff
@@ -1019,20 +1019,55 @@ void computer_session::action_srcf_seal()
 void computer_session::action_srcf_elevator()
 {
     Character &player_character = get_player_character();
-    if( !player_character.has_amount( itype_sarcophagus_access_code, 1 ) ) {
-        print_error( _( "Access code required!" ) );
-    } else {
-        player_character.use_amount( itype_sarcophagus_access_code, 1 );
+    map &here = get_map();
+    tripoint surface_elevator;
+    tripoint underground_elevator;
+    bool is_surface_elevator_on;
+    bool is_underground_elevator_on;
```

Comment:
Getting compiler warnings of uninitialised variables. Could initialise with some sentry value, and in an erroneous case where no elevator control is found, can jump a `debugmsg` and return before screwing up the map.

Reply:
Initialized booleans (no default values? wow), regarding the latter I added check, which can print an error message to the in-game console.

Reply:
A corner case came up to my mind when reading the code is when, for whatever strange reason, there is neither `t_elevator_control_off` nor `t_elevator_control` existing, then these four uninitialised local variables are never assigned so they have random dirty value and may screw up the next part of the code.

Reply:
I got this idea too when I realized I need to initialize booleans.
It would enter the if and could try to change some random point on the map.